sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/util/CosmosPagedFlux.java

```diff
@@ -128,19 +132,19 @@ public void subscribe(CoreSubscriber<? super T> coreSubscriber) {
         final AtomicReference<Context> parentContext = new AtomicReference<>(Context.NONE);
         AtomicReference<Instant> startTime = new AtomicReference<>();
         return this.optionsFluxFunction.apply(pagedFluxOptions).doOnSubscribe(ignoredValue -> {
-            if (pagedFluxOptions.getTracerProvider().isEnabled()) {
+            if (pagedFluxOptions.getTracerProvider() != null && pagedFluxOptions.getTracerProvider().isEnabled()) {
```

Comment:
I would create a helper method to avoid repeating the code

```suggestion
            if (isTracerEnabled()) {
```

and define the helper method:
```java
boolean isTracerEnabled() {
  return this.pagedFluxOptions.getTracerProvider() != null && pagedFluxOptions.getTracerProvider().isEnabled()
```

Reply:
its just two condition so should be fine, if it increase i will add in future